core/src/main/scala/org/http4s/parser/LinkHeader.scala

```diff
@@ -1,35 +1,52 @@
 package org.http4s
 package parser
 
-import org.http4s.headers.Link
+import java.nio.charset.StandardCharsets
+
+import org.http4s.headers.{Link, LinkValue}
 import org.http4s.internal.parboiled2.support.{::, HNil}
-import org.http4s.internal.parboiled2.{Rule, Rule1}
+import org.http4s.internal.parboiled2._
+
+private[parser] trait LinkHeader {
+  def LINK(value: String): ParseResult[Link] =
+    new LinkParser(value).parse
 
-trait LinkHeader {
-  def LINK(value: String): ParseResult[Link] = new LinkParser(value).parse
+  private class LinkParser(input: ParserInput)
+      extends Http4sHeaderParser[Link](input)
+      with MediaRange.MediaRangeParser
+      with Rfc3986Parser {
 
-  private class LinkParser(value: String)
-      extends UriHeaderParser[Link](value)
-      with MediaRange.MediaRangeParser {
+    override def charset: java.nio.charset.Charset = StandardCharsets.ISO_8859_1
 
-    override def fromUri(uri: Uri): Link = Link(uri)
+    def entry: Rule1[Link] = rule {
+      oneOrMore(LinkValueWithAttr).separatedBy(ListSep) ~> { links: Seq[LinkValue] =>
+        Link(links.head, links.tail: _*)
+      }
+    }
+
+    def LinkValueWithAttr: Rule1[LinkValue] = rule {
+      "<" ~ LinkValue ~ ">" ~ zeroOrMore(";" ~ OptWS ~ LinkValueAttr)
+    }
 
-    override def entry: Rule1[Link] = rule {
-      "<" ~ super[UriHeaderParser].entry ~ ">" ~ zeroOrMore(";" ~ OptWS ~ LinkAttr)
+    def LinkValue: Rule1[LinkValue] = rule {
```

Comment:
Taken from `UriHeaderParser` (along with the `charset` member). Tried extending `UriHeaderParser` instead of the base trait `Http4sHeaderParser`, but I had some compile issues -- possibly due to the order in which traits were implemented.